libsolutil/CommonData.h

```diff
@@ -232,6 +232,39 @@ std::set<K> keys(std::map<K, V> const& _map)
 	return applyMap(_map, [](auto const& _elem) { return _elem.first; }, std::set<K>{});
 }
 
+/// @returns a pointer to the entry of @a _map at @a _key, if there is one, and nullptr otherwise.
+template<typename MapType, typename KeyType>
+decltype(auto) valueOrNullptr(MapType&& _map, KeyType const& _key)
+{
+	auto it = _map.find(_key);
+	return (it == _map.end()) ? nullptr : &it->second;
+}
+
+namespace detail
+{
+struct allow_copy {};
+}
+static constexpr auto allow_copy = detail::allow_copy{};
```

Comment:
Not sure how you guys feel about this construct - but neither generally allowing it to create copies, nor generally disallowing it, strikes me as a good way to deal with this, whereas forcing to make copies explicit with a nice readable error message, if one doesn't, seems ok to me.